### DISCUSSION TITLE
Approval Timeline>

### DISCUSSION THREAD

**Discussion by ghost** [amzn/selling-partner-api-models#3715] [General]
If we selected all non-PII roles, how long does it typically take for a public app to be approved?

We were initially approved as a private application with a couple roles, but then added all and switched to public.

**Comment by MeenaAmz** [comment]
Please refer to the process in [here](https://developer-docs.amazon.com/sp-api/docs/register-as-a-public-developer). It usually takes a few days to get a response on your case. Look out for the case updates and respond to any queries promptly.
